frontend/public/components/storage-class-form.tsx

```diff
@@ -0,0 +1,927 @@
+      }
+
+      this.previousName = updatedName;
+    }
+
+    return returnVal;
+  };
+
+  validateParameters = () => {
+    let validationSuccessful = true;
+    const params = this.state.newStorageClass.parameters;
+    for (let key in params) {
+      if (params.hasOwnProperty(key) && params[key].validationMsg !== null) {
+        validationSuccessful = false;
+        break;
+      }
+    }
+
+    return validationSuccessful;
+  };
+
+  allRequiredFieldsFilled = () => {
+    if (this.state.newStorageClass.name.trim().length === 0) {
+      return false;
+    }
+
+    const storageType = this.state.newStorageClass.type;
+    if (storageType === null) {
+      return false;
+    }
+
+    const allParamsForType = this.storageTypes[storageType].parameters;
+    const userEnteredParams = this.state.newStorageClass.parameters;
+    let allReqdFieldsEntered = true;
+
+    for (let key in allParamsForType) {
+      if (allParamsForType.hasOwnProperty(key)) {
+        const value = _.get(userEnteredParams, `${key}.value`, null);
+        const valueExists = value !== null && value !== '' && !_.isEmpty(value);
+
+        if (this.paramIsRequired(key) && !valueExists) {
+          allReqdFieldsEntered = false;
+          break;
+        }
+      }
+    }
+
+    return allReqdFieldsEntered;
+  };
+
+  paramIsRequired = (paramKey, params = this.state.newStorageClass.parameters) => {
+    const requiredParam = _.get(this.storageTypes[this.state.newStorageClass.type], ['parameters', paramKey, 'required'], null);
+    let isRequired = false;
+    if (requiredParam) {
+      isRequired = typeof requiredParam === 'function' ? requiredParam(params) : requiredParam;
+    }
+
+    return isRequired;
+  };
+
+  getProvisionerElements = () => {
+    const dynamicContent = Object.keys(this.storageTypes[this.state.newStorageClass.type].parameters).map(key => {
+      const parameter = this.storageTypes[this.state.newStorageClass.type].parameters[key];
+      const validationMsg = _.get(this.state.newStorageClass.parameters[key], 'validationMsg', null);
+      const isCheckbox = parameter.type === 'checkbox';
+
+      if (parameter.visible && !parameter.visible(this.state.newStorageClass.parameters)) {
+        return null;
+      }
+
+      let children =
+        <React.Fragment>
+          <FormControl
+            type={isCheckbox ? 'checkbox': 'text'}
+            className={isCheckbox ? 'create-storage-class-form__checkbox' : ''}
+            checked={_.get(this.state, `newStorageClass.parameters.${key}.value`, false)}
+            value={_.get(this.state, `newStorageClass.parameters.${key}.value`, '')}
+            onChange={(event) => this.setParameterHandler(key, event, isCheckbox)} />
+          <HelpBlock>{validationMsg ? validationMsg : parameter.placeholder}</HelpBlock>
+        </React.Fragment>;
+
+      if (parameter.values) {
+        children =
+          <React.Fragment>
+            <Dropdown
+              title={parameter.placeholder}
```

Comment:
If a value is selected, I believe this should be set to the title of the selected value. (The `Dropdown` component is confusing, I know.)